summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/VersionedBatchStore.scala

```diff
@@ -200,13 +201,19 @@ class VersionedBatchStore[K, V, K2, V2](rootPath: String, versionsToKeep: Int, o
       case _ => Some(batchVersion)
     }
 
-    lastVals.map(pack(batchID, _))
-      .toPipe((0, 1))
-      .write(VersionedKeyValSource[K2, V2](rootPath,
-        sourceVersion = None,
-        sinkVersion = newVersion,
-        maxFailures = 0,
-        versionsToKeep = versionsToKeep))
+    val target = VersionedKeyValSource[K2, V2](rootPath,
+      sourceVersion = None,
+      sinkVersion = newVersion,
+      maxFailures = 0,
+      versionsToKeep = versionsToKeep)
+    if (!target.sinkExists(mode)) {
+      logger.info(s"Versioned batched store version for $this @ $newVersion doesn't exist. Will write out.")
+      lastVals.map(pack(batchID, _))
+        .write(target)
+    } else {
+      logger.warn(s"""Versioned batched store version for $this @ $newVersion already exists! Will skip adding to plan.
+        | Note you must have at least one store that requires writing out or the job will fail.""".stripMargin('|'))
```

Comment:
How will it recover? Imagine we write the store but crash before we write to statebird. Statebird keeps telling you to run yesterday's block, but the job keeps failing because it has already written and there is only one store.